File: src/main/java/hu/frontrider/gearcraft/blocks/CreativeGearbox.java
package hu.frontrider.gearcraft.blocks;

import hu.frontrider.gearcraft.api.IPoweredBlock;
import net.minecraft.block.Block;
import net.minecraft.block.material.MapColor;
import net.minecraft.block.material.Material;
import net.minecraft.util.EnumFacing;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockAccess;

import java.util.List;

import static hu.frontrider.gearcraft.GearCraft.MODID;

public class CreativeGearbox  extends Block implements IPoweredBlock,TooltippedBlock {
    private static final String blockName = "creative_gearbox";

    public CreativeGearbox() {
        super(Material.ROCK,MapColor.BLACK);
        setBlockUnbreakable();
        setRegistryName(MODID,blockName);
        setUnlocalizedName(blockName);
    }

    @Override
    public int getPower(IBlockAccess iBlockAccess, BlockPos blockPos) {
        return 4;
    }

    @Override
    public int getStrength(IBlockAccess iBlockAccess, BlockPos blockPos) {
        return Integer.MAX_VALUE;
    }

    @Override
    public boolean isValidSide(IBlockAccess access, BlockPos pos, EnumFacing facing) {
        return true;
    }

    @Override
    public void setTooltip(List<String> tooltip) {
        tooltip.add("Infinite power");
        tooltip.add("Creative Only");
    }
}

File: mod/src/main/java/hu/frontrider/gearcraft/blocks/CreativeGearbox.kt
package hu.frontrider.gearcraft.blocks

import hu.frontrider.gearcraft.api.IPoweredBlock
import net.minecraft.block.Block
import net.minecraft.block.material.MapColor
import net.minecraft.block.material.Material
import net.minecraft.util.EnumFacing
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockAccess

import hu.frontrider.gearcraft.GearCraft.Companion.MODID

class CreativeGearbox : Block(Material.ROCK, MapColor.BLACK), IPoweredBlock, TooltippedBlock {
    init {
        setRegistryName(MODID, blockName)
        unlocalizedName = "$MODID.$blockName"
    }

    override fun getPower(iBlockAccess: IBlockAccess, blockPos: BlockPos): Int {
        return 4
    }

    override fun getStrength(iBlockAccess: IBlockAccess, blockPos: BlockPos): Int {
        //the closest we can get to infinity
        return Integer.MAX_VALUE
    }

    override fun isValidSide(access: IBlockAccess, pos: BlockPos, facing: EnumFacing): Boolean {
        return true
    }

    override fun setTooltip(tooltip: MutableList<String>) {
        tooltip.add("Infinite power")
        tooltip.add("Creative Only")
    }

    companion object {
        private val blockName = "creative_gearbox"
    }
}
